separate_factors.py

```diff
@@ -0,0 +1,181 @@
+"""
+Generate separate text and factor files from the combined data
+"""
+
+import os
+import argparse
+import logging
+logging.basicConfig(level=logging.INFO,
+                    format='[%(asctime)s][%(levelname)s] %(message)s',
+                    datefmt='%Y-%m-%d %H:%M:%S')
+
+
+SRC_LANG = 'de'
+TGT_LANG = 'en'
+SEGMENTS_SUFFIX = 'segments'
+
+SPLITS = ['train', 'valid', 'test']
+
+PAUSE_TOKEN = '[pause]'
+EOW_TOKEN = '<eow>'
+NOT_EOW_TOKEN = '<!eow>'
+EOW_FACTOR_SPECIAL_TOKEN = '<special>'
+DUMMY_DURATION = 0  # Duration value for eow/pause
+SRC_DURATION_SEP = '<||>'
+
+FACTOR_TYPES = [
+    'DURATION',
+    'PAUSES_REMAINING',
+    'TOTAL_DURATION_REMAINING',
+    'SEGMENT_DURATION_REMAINING',
+    'EOW'
+]
+
+class FactorFileManager:
+    """
+    Context Manager to neatly manage all the factor output file handles
+    """
+    def __init__(self, prefix):
+        self.handles = dict()
+        self.prefix = prefix
+
+    def __enter__(self):
+        for f in FACTOR_TYPES:
+            output_path = self.prefix + f.lower()
+            self.handles[f] = open(output_path, 'w')
```

Comment:
can you check if file exists already? and if it does either add a warning before opening in write mode or skip writing to it at all (assuming the file is correct).

Reply:
Skipping individual files is trickier, but changed it to raise `FileExistsError` if file exists already, with a `--allow-overwrite` option to explicitly allow overwriting